pimd/mtracebis_netlink.c

```diff
@@ -0,0 +1,700 @@
+/*
```

Comment:
I don't think this belongs here, for a few reasons.

1. If we were to copy iproute2 library code, it would go in `lib/` and not here.
2. Zebra is the interface to the kernel, I don't believe any other daemons with the exception of nhrp and the fpm module speak Netlink. We should think about how we want to organize the communication this module does.

Reply:
OK. I must admit I was not completely comfortable about using this library.

But I could not figure out how to pass routing information from pimd...

I thought of writing it as a pimd command but I was under the impression that vty does not update until the command is finished running. I could be wrong, but I tried a command that used vty_out and sleeps and nothing printed on vtysh. Also, I am not sure it is possible to
interrupt a native command. So, I went for the route of independent program which can also be interrupted from vtysh. But vtysh does not know about multicast next-hop lookup, which is needed to determine out of which interface to send the query, hence I needed a way to
get outgoing interface for particular source... So you see, it is complicated.

It could be a problem using the netlink interface, if the query this program does returns a whole chunk of database...

There is the option of doing the interface selection manually...

So, you type e.g. mtrace eth0 <mc source>. Then the program is less clever but you avoid using the netlink lib.

Maybe that is a conservative safe option?

Reply:
You can ignore my comment, I misunderstood the design of this program. I am fine with the design as it is.